src/loader/mod.rs

```diff
@@ -525,11 +569,133 @@ fn align_up(addr: u64, align: u64) -> usize {
     }
 }
 
+/// Writes the device tree to the given memory slice.
+///
+/// # Arguments
+///
+/// * `guest_mem` - A u8 slice that will be partially overwritten by the device tree blob.
+/// * `guest_addr` - The address in `guest_mem` at which to load the device tree blob.
+/// * `dtb_image` - The device tree blob.
+#[cfg(target_arch = "aarch64")]
+pub fn load_dtb<F, M: GuestMemory>(
+    guest_mem: &M,
+    guest_addr: GuestAddress,
+    dtb_image: &mut F,
+) -> Result<()>
+where
+    F: Read + Seek,
+{
+    let dtb_size = dtb_image
+        .seek(SeekFrom::End(0))
+        .map_err(|_| Error::SeekDtbEnd)? as usize;
+    if dtb_size > 0x200000 {
+        return Err(Error::DtbTooBig);
+    }
+    dtb_image
+        .seek(SeekFrom::Start(0))
+        .map_err(|_| Error::SeekDtbStart)?;
+    guest_mem
+        .read_exact_from(guest_addr, dtb_image, dtb_size)
+        .map_err(|_| Error::ReadDtbImage)
+}
+
+#[cfg(feature = "image")]
+#[cfg(target_arch = "aarch64")]
+/// ARM64 Image (PE) format support
+pub struct Image;
+
+#[cfg(feature = "image")]
+#[cfg(target_arch = "aarch64")]
+#[repr(C)]
+#[derive(Debug, Copy, Clone, Default)]
+// See kernel doc Documentation/arm64/booting.txt for more information.
+struct arm64_image_header {
+    code0: u32,
+    code1: u32,
+    text_offset: u64,
+    image_size: u64,
+    flags: u64,
+    res2: u64,
+    res3: u64,
+    res4: u64,
+    magic: u32,
+    res5: u32,
+}
+
+#[cfg(feature = "image")]
+#[cfg(target_arch = "aarch64")]
+impl KernelLoader for Image {
+    /// Loads a Image
+    ///
+    /// # Arguments
+    ///
+    /// * `guest_mem` - The guest memory where the kernel image is loaded.
+    /// * `kernel_start` - The offset into 'guest_mem' at which to load the kernel.
+    /// * `kernel_image` - Input Image image.
+    /// * `highmem_start_address` - ignored on ARM64
+    ///
+    /// # Returns
+    /// * KernelLoaderResult
```

Comment:
nit: I would remove this. I know it is the same as for Elf, but I would remove that as well (I'll open an issue with that). The fact that it returns a `KernelLoaderResult` can be seen from the function signature where you can also click on KernelLoaderResult and expend it.

Reply:
I'm somewhat tracking that in #23 too. Would have fixed this occurrence as well as part of the rebase.

Reply:
Ok, let's fix it in the other PR then.